src/System.IO.FileSystem/src/System/IO/WinRTFileSystem.cs

```diff
@@ -227,14 +228,86 @@ public override bool DirectoryExists(string fullPath)
 
         public override IEnumerable<string> EnumeratePaths(string fullPath, string searchPattern, SearchOption searchOption, SearchTarget searchTarget)
         {
-            // Fill in implementation
-            return new string[0];
+            IEnumerable<IStorageItem> storageFiles = SynchronousResultOf(EnumerateFileQuery(fullPath, searchPattern, searchOption, searchTarget));
+
+            // Yield the paths of the StorageItems
+            foreach (IStorageItem item in storageFiles)
+            {
+                yield return item.Path;
+            }
         }
 
         public override IEnumerable<FileSystemInfo> EnumerateFileSystemInfos(string fullPath, string searchPattern, SearchOption searchOption, SearchTarget searchTarget)
         {
-            // Fill in implementation
-            return new FileSystemInfo[0];
+            IEnumerable<IStorageItem> storageFiles = SynchronousResultOf(EnumerateFileQuery(fullPath, searchPattern, searchOption, searchTarget));
+
+            // Translate the StorageFile results into FileSystemInfos
+            foreach (IStorageItem item in storageFiles)
+            {
+                if (item.IsOfType(StorageItemTypes.Folder))
+                {
+                    yield return new DirectoryInfo(item.Path);
+                }
+                else // If it is neither a File nor folder then we treat it as a File.
+                {
+                    yield return new FileInfo(item.Path);
```

Comment:
Potential future perf optimization: The `IStorageItem` could potentially be passed along to the `FileInfo`/`DirectoryInfo` so that it doesn't have to be recreated again when the properties are accessed. WinRT-specific internal constructors would have to be added that would create an instance of [`WinRTFileSystemObject`](https://github.com/dotnet/corefx/blob/d0dc5fc099946adc1035b34a8b1f6042eddb0c75/src/System.IO.FileSystem/src/System/IO/WinRTFileSystemObject.cs#L22) from the `IStoreItem`, and then set the [WinRT FileSystemInfo's `IFileSystemObject`](https://github.com/dotnet/corefx/blob/dad7638e417019f9173f6bbd136bc5a5cd833f4a/src/System.IO.FileSystem/src/System/IO/FileSystemInfo.WinRT.cs#L9) field to that instance when the `FileInfo`/`DirectoryInfo` is created.